packages/design-system-react-native/src/temp-components/Spinner/Spinner.types.ts

```diff
@@ -0,0 +1,31 @@
+import type { ViewProps } from 'react-native';
+
+import type { IconColor, IconProps } from '../../components/Icon';
+import type { TextProps } from '../../components/Text/Text.types';
+
+/**
+ * Spinner component props.
+ */
+export type SpinnerProps = {
+  /**
+   * Optional prop that sets the color of the spinner icon using predefined theme colors
+   * @default IconColor.IconDefault
+   */
+  color?: IconColor;
+  /**
+   * Optional props to be passed to the loadingText element
+   */
+  spinnerIconProps?: Partial<IconProps>;
```

Comment:
Oh this is good to know I was creating a custom type to make props optional. Created an issue to replace the custom one with Partial in DSR https://github.com/MetaMask/metamask-design-system/issues/356